src/module.cpp

```diff
@@ -533,11 +533,129 @@ void Module::AddGlobalVariable(const std::string &name, const Type *type, Expr *
             initExpr = TypeCheck(initExpr);
             if (initExpr != nullptr) {
                 // We need to make sure the initializer expression is
-                // the same type as the global.  (But not if it's an
-                // ExprList; they don't have types per se / can't type
-                // convert themselves anyway.)
-                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr)
+                // the same type as the global.
+                if (llvm::dyn_cast<ExprList>(initExpr) == nullptr) {
                     initExpr = TypeConvertExpr(initExpr, type, "initializer");
+                } else {
+                    // The alternative is to create initializer expression with
+                    // correct type and value from ExprList.
+                    // If we have exprList that initalizes const varying int, e.g.:
+                    // static const int x = { 0, 1, 2, 3 };
+                    // then we need to convert initExpr to constExpr that can
+                    // be utilized later in arithmetic expressions that can be
+                    // calculated in compile time (Optimize function call below),
+                    ExprList *exprList = llvm::dyn_cast<ExprList>(initExpr);
+                    if (type->IsConstType() && type->IsVaryingType() && type->IsAtomicType()) {
+                        const AtomicType::BasicType basicType = CastType<AtomicType>(type)->basicType;
+                        const int N = g->target->getVectorWidth();
+                        bool *vals_bool = (bool *)alloca(N * sizeof(bool));
+                        for (int j = 0; j < N; j++) {
+                            vals_bool[j] = false;
+                        }
```

Comment:
```suggestion
                        std::vector<bool> vals_bool(N, false);
```

Reply:
No, it won't work well with `ConstExpr::ConstExpr(const Type *t, bool *b, SourcePos p)`. Because, `std::vector<bool>` is a specialization of vector that does not necessarily store its elements as a contiguous array ([see here](https://en.cppreference.com/w/cpp/container/vector_bool)), so `data` is deleted (or returns undereferenceable `void`) ([godbolt example](https://godbolt.org/z/7qhes35Ej)).